### PR TITLE
Adding threadwisegemm lowering implementation

### PR DESCRIPTION
**Pull request by jerryyin**
Follow-up to #13

### CODE REVIEW

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 1]
```diff
@@ -1507,7 +1508,29 @@ struct BlockwiseGemmRewritePattern : public OpRewritePattern<miopen::BlockwiseGe
...
     //                                        decltype(b_thread_mtx),
     //                                        decltype(c_thread_mtx)>{};
-    lb.create<miopen::ThreadwiseGemmOp>(op.getLoc(), threadAAllocOp, threadBAllocOp, op.getOperand(2));
+
+    // TODO: Update hard-coded gemmA/B/C shape to actual shapes when
+    // threadwisecopy done
+    ArrayRef<int64_t> threadAShape = {8, 16};
+    auto threadA2DMemRefType =
+        MemRefType::get(threadAShape, b.getF32Type(), {}, registerMemorySpace);
+    auto threadA2DViewOp = lb.create<ViewOp>(op.getLoc(), threadA2DMemRefType,
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: @jerryyin do we really need these ViewOp?

Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: I can get rid of them. The threadwisegemm lowering op is assuming 2d dimension and the viewOp is just a way to feed fake 2d memref. Once you have the dimension right, I can do a rebase and get rid of the ViewOp.

**Review by whchung — CHANGES_REQUESTED** [review]
@jerryyin

1) please remove those ViewOps. I'll amend BlockwiseGemm implementation to make sure the dimensions are correct.

2) also could you specify commands to run `mlir-opt` so we can lower affine dialect introduced in this patch all the way to std

3) in a subsequent PR, could you modify "-convert-miopen-to-gpu" to cover steps needed in 2).

**Comment by whchung** [comment]
@jerryyin to the very least I want this to work:

```
./bin/mlir-miopen-driver -p | ./bin/mlir-opt -miopen-lowering -miopen-affine-transform -miopen-affix-params -miopen-lowering-step2  -miopen-lowering-step3
```

and please specify what steps after `-miopen-lowering-step3` are needed to convert affine to std

**Comment by jerryyin** [comment]
To convert to affine:

`./bin/mlir-miopen-driver -p | ./bin/mlir-opt -miopen-lowering -miopen-affine-transform -miopen-affix-params -miopen-lowering-step2  -miopen-lowering-step3`

To convert to std:

`./bin/mlir-miopen-driver -p | ./bin/mlir-opt -miopen-lowering -miopen-affine-transform -miopen-affix-params -miopen-lowering-step2  -miopen-lowering-step3 --lower-affine`

I will draft another PR to add `--lower-affine` to `-convert-miopen-to-gpu` pass.

**Review by whchung — APPROVED** [review]

**Pull request merged**